hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if values.get('holiday_status_id'):
+            type_obj = self.env['hr.holidays.status']
+            type = type_obj.browse(values.get('holiday_status_id'))
+            if type.auto_approve:
+                result.sudo().holidays_validate()
```

Comment:
The holidays_validate is a method called by the workflow.
Please trigger workflow processes, instead of calling the method directly.

Also, are you sure this always works ? The `admin` user always have rights to call `holidays_validate` ?
How about checking the `auto_approve` status in the `holidays_validate` (or workflow condition) method instead ?
This way, you don't need sudo, so the write_uid will be consistent.